http/events.go

```diff
@@ -163,3 +171,45 @@ func handleEventsSubscribe(core *vault.Core, req *logical.Request) http.Handler
 		}
 	})
 }
+
+// validateNamespacePatterns validates that the namespace subscribe patterns only match
+// child namespaces of the request namespace. It also checks if the request namespace
+// is present, and if it is not, it adds it. It returns a (possibly modified) copy of
+// the patterns.
+func validateNamespacePatterns(patterns []string, requestNamespace *namespace.Namespace) ([]string, error) {
+	rns := strings.TrimSuffix(requestNamespace.Path, "/")
+	for _, pattern := range patterns {
+		if !isChildPattern(pattern, rns) {
+			return nil, fmt.Errorf("namespace pattern %v must match only child namespaces of %s", pattern, rns)
+		}
+	}
+
+	// check that the request namespace is included in at least one of the patterns
+	present := false
+	for _, pattern := range patterns {
+		if glob.Glob(pattern, rns) {
+			present = true
+			break
+		}
+	}
+	if present {
+		return patterns, nil
+	}
+	newPatterns := append(patterns, rns)
+	return newPatterns, nil
+}
+
+func isChildPattern(pattern string, ns string) bool {
```

Comment:
Discussed in private but just commenting here too to help me track: We agreed to tweak this slightly so clients provide child namespaces relative to the request's own namespace. In that case, we won't need to "check" that a pattern is a child namespace, because we'll just automatically prepend the request namespace to the path anyway.

Reply:
Done! Definitely is simpler to reason about.